shared/reducers/config.js

```diff
@@ -3,51 +3,107 @@ import logger from '../logger'
 import * as I from 'immutable'
 import * as Types from '../constants/types/config'
 import * as Constants from '../constants/config'
+import * as ChatConstants from '../constants/chat2'
 import * as ConfigGen from '../actions/config-gen'
 
 const initialState = Constants.makeState()
 
 export default function(
   state: Types.State = initialState,
-  action:
-    | ConfigGen.Actions
-    | ConfigGen.ChangedFocusPayload
-    | ConfigGen.ChangedActivePayload
-    | {type: 'remote:updateMenubarWindowID', payload: {id: number}}
+  action: ConfigGen.Actions | {type: 'remote:updateMenubarWindowID', payload: {id: number}}
 ): Types.State {
   switch (action.type) {
     case ConfigGen.resetStore:
-      return initialState
-        .set('readyForBootstrap', state.readyForBootstrap)
-        .set('menubarWindowID', state.menubarWindowID)
+      return initialState.merge({
+        appFocused: state.appFocused,
+        appFocusedCount: state.appFocusedCount,
+        configuredAccounts: state.configuredAccounts,
+        daemonHandshakeState: state.daemonHandshakeState,
+        daemonHandshakeWaiters: state.daemonHandshakeWaiters,
+        defaultUsername: state.defaultUsername,
+        logoutHandshakeWaiters: state.logoutHandshakeWaiters,
+        menubarWindowID: state.menubarWindowID,
+        pushLoaded: state.pushLoaded,
+        startupDetailsLoaded: state.startupDetailsLoaded,
+      })
+    case ConfigGen.restartHandshake:
+      return state.merge({
+        daemonError: null,
+        daemonHandshakeFailedReason: '',
+        daemonHandshakeRetriesLeft: Math.max(state.daemonHandshakeRetriesLeft - 1, 0),
+        daemonHandshakeState: 'starting',
+      })
+    case ConfigGen.startHandshake:
+      return state.merge({
+        daemonError: null,
+        daemonHandshakeFailedReason: '',
+        daemonHandshakeRetriesLeft: Constants.maxHandshakeTries,
+        daemonHandshakeState: 'starting',
+      })
+    case ConfigGen.logoutHandshake:
+      return state.merge({logoutHandshakeWaiters: I.Map()})
+    case ConfigGen.daemonHandshake:
+      return state.set('daemonHandshakeState', 'waitingForWaiters')
```

Comment:
you changed a bunch of these to `merge` below, this one too?

Reply:
k